components/kyma-environment-broker/cmd/broker/upgrade_kyma_test.go

```diff
@@ -86,3 +88,125 @@ func TestKymaUpgrade_UpgradeTo2(t *testing.T) {
 	found := suite.provisionerClient.IsRuntimeUpgraded(upgradeOp.InstanceDetails.RuntimeID, "2.0.0-rc4")
 	assert.False(t, found)
 }
+
+func TestKymaUpgrade_UpgradeAfterMigration(t *testing.T) {
+	// given
+	suite := NewBrokerSuiteTest(t)
+	mockBTPOperatorClusterID()
+	defer suite.TearDown()
+	id := "InstanceID-UpgradeAfterMigration"
+
+	// provision Kyma 2.0
+	resp := suite.CallAPI("PUT", fmt.Sprintf("oauth/cf-eu10/v2/service_instances/%s?accepts_incomplete=true", id), `
+{
+	"service_id": "47c9dcbf-ff30-448e-ab36-d3bad66ba281",
+	"plan_id": "4deee563-e5ec-4731-b9b1-53b42d855f0c",
+	"context": {
+		"sm_platform_credentials": {
+			"url": "https://sm.url",
+			"credentials": {
+			"basic": {
+					"username":"smUsername",
+					"password":"smPassword"
+	  			}
+			}
+		},
+		"globalaccount_id": "g-account-id",
+		"subaccount_id": "sub-id",
+		"user_id": "john.smith@email.com"
+	},
+	"parameters": {
+		"name": "testing-cluster",
+		"kymaVersion": "2.0.0-rc4"
+	}
+}`)
+	opID := suite.DecodeOperationID(resp)
+	suite.processReconcilingByOperationID(opID)
+	suite.WaitForOperationState(opID, domain.Succeeded)
+
+	// migrate svcat
+	resp = suite.CallAPI("PATCH", fmt.Sprintf("oauth/cf-eu10/v2/service_instances/%s?accepts_incomplete=true", id), `
+{
+	"service_id": "47c9dcbf-ff30-448e-ab36-d3bad66ba281",
+	"context": {
+		"globalaccount_id": "g-account-id",
+		"subaccount_id": "sub-id",
+		"user_id": "john.smith@email.com",
+		"sm_operator_credentials": {
+			"clientid": "testClientID",
+			"clientsecret": "testClientSecret",
+			"sm_url": "https://service-manager.kyma.com",
+			"url": "https://test.auth.com",
+			"xsappname": "testXsappname"
+		},
+		"isMigration": true
+	}
+}`)
+
+	// make sure migration finished
+	assert.Equal(t, http.StatusAccepted, resp.StatusCode)
+	updateOperationID := suite.DecodeOperationID(resp)
+	suite.FinishUpdatingOperationByProvisioner(updateOperationID)
+	suite.FinishUpdatingOperationByReconcilerBoth(updateOperationID)
+	suite.WaitForOperationState(updateOperationID, domain.Succeeded)
+
+	// run upgrade
+	orchestrationResp := suite.CallAPI("POST", "upgrade/kyma", `
+{
+	"strategy": {
+		"type": "parallel",
+		"schedule": "immediate",
+		"parallel": {
+			"workers": 1
+		}
+	},
+	"dryRun": false,
+	"targets": {
+		"include": [
+			{
+				"subAccount": "sub-id"
+			}
+		]
+	},
+	"kyma": {
+		"kymaVersion": "2.0.0"
```

Comment:
in this test, I am trying to upgrade kyma from `2.0.0-rc4` to `2.0.0`. I observe `orchestration` getting created, this creates `upgrade` operation for this particular one SKR, however, the version it tries to upgrade to is `1.24.7`.

I think the `RuntimeVersion.ForUpgrade` is probably missing something. As far as I can tell, the orchestration didn't create AccountMapping with new version for the globalaccount/subaccount, the upgrade_kyma queue is trying to use default instead of what is defined in the upgrade payload.

https://github.com/wozniakjan/control-plane/blob/d1a52ea048d743d3cd6a48a640e430537b72e144/components/kyma-environment-broker/internal/runtimeversion/runtimeversion.go#L112-L127